src/xdealer_channel.cpp

```diff
@@ -0,0 +1,81 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "xdearler_channel.hpp"
+
+namespace xeus
+{
+
+    xdearler_channel::xdearler_channel(zmq::context_t& context,
+                                 const std::string& user_name,
+                                 const xeus::xconfiguration& config)
+        : m_shell(context, zmq::socket_type::dealer)
+        , m_control(context, zmq::socket_type::dealer)
+        , m_shell_end_point("")
+        , m_control_end_point("")
+        , m_user_name(user_name)
+        , m_session_id(xeus::new_xguid())
```

Comment:
I don't think you need `user_name` and `session_id` here.

Reply:
Yeah correct, It was present in the initial design. I don't think we shall we needing `user_name` or `session_id` anywhere here. Getting rid of these.